azure_functions_worker/dispatcher.py

```diff
@@ -330,10 +333,12 @@ async def _handle__invocation_request(self, req):
                     trigger_metadata = invoc_request.trigger_metadata
                 else:
                     trigger_metadata = None
+
                 args[pb.name] = bindings.from_incoming_proto(
-                    pb_type_info.binding_name, pb.data,
+                    pb_type_info.binding_name, pb,
                     trigger_metadata=trigger_metadata,
-                    pytype=pb_type_info.pytype)
+                    pytype=pb_type_info.pytype,
+                    shmem_mgr=self._shmem_mgr)
```

Comment:
This thing of passing the shared memory manager all over (also applies to the C# code) is getting a little repetitive.
Isn't there a better option? Putting it in a context? Making it a singleton?

Reply:
I don't see use of globals/singletons in this repo so not sure if that's something people would want. But we can get input on that once we put this in review.